definitions/subworkflows/single_sample_sv_callers.cwl

```diff
@@ -0,0 +1,141 @@
+    reference:
+        type: string
+
+    cnvkit_access:
+        type: File?
+    cnvkit_bait_intervals:
+        type: File?
+    cnvkit_diagram:
+        type: boolean?
+    cnvkit_drop_low_coverage:
+        type: boolean?
+    cnvkit_method:
+        type: string?
+    cnvkit_reference_cnn:
+        type: File?
+    cnvkit_scatter_plot:
+        type: boolean?
+
+    manta_call_regions:
+        type: File?
+    manta_non_wgs:
+        type: boolean?
+    manta_output_contigs:
+        type: boolean?
+
+    smoove_exclude_regions:
+        type: File?
+
+outputs:
+    intervals_antitarget:
+        type: File?
+        outputSource: run_cnvkit/intervals_antitarget
+    intervals_target:
+        type: File?
+        outputSource: run_cnvkit/intervals_target
+    normal_antitarget_coverage:
+        type: File?
+        outputSource: run_cnvkit/normal_antitarget_coverage
+    normal_target_coverage:
+        type: File?
+        outputSource: run_cnvkit/normal_target_coverage
+    reference_coverage:
+        type: File?
+        outputSource: run_cnvkit/reference_coverage
+    cn_diagram:
+        type: File?
+        outputSource: run_cnvkit/cn_diagram
+    cn_scatter_plot:
+        type: File?
+        outputSource: run_cnvkit/cn_scatter_plot
+    tumor_antitarget_coverage:
+        type: File
+        outputSource: run_cnvkit/tumor_antitarget_coverage
+    tumor_target_coverage:
+        type: File
+        outputSource: run_cnvkit/tumor_target_coverage
+    tumor_bin_level_ratios:
+        type: File
+        outputSource: run_cnvkit/tumor_bin_level_ratios
+    tumor_segmented_ratios:
+        type: File
+        outputSource: run_cnvkit/tumor_segmented_ratios
+    manta_diploid_variants:
+        type: File?
+        outputSource: run_manta/diploid_variants
+    manta_somatic_variants:
+        type: File?
+        outputSource: run_manta/somatic_variants
+    manta_all_candidates:
+        type: File
+        outputSource: run_manta/all_candidates
+    manta_small_candidates:
+        type: File
+        outputSource: run_manta/small_candidates
+    manta_tumor_only_variants:
+        type: File?
+        outputSource: run_manta/tumor_only_variants
+    smoove_output_variants:
+        type: File
+        outputSource: run_smoove/output_vcf
+
+steps:
+    cram_to_bam:
+        run: ../tools/cram_to_bam.cwl
+        in:
+            cram: cram
+            reference: reference
+        out:
+            [bam]
+    index_bam:
+        run: ../tools/index_bam.cwl
+        in:
+            bam: cram_to_bam/bam
+        out:
+            [indexed_bam]
+    run_cnvkit:
+        run: ../tools/cnvkit_batch.cwl
+        in:
+            access: cnvkit_access
+            bait_intervals: cnvkit_bait_intervals
+            diagram: cnvkit_diagram
+            drop_low_coverage: cnvkit_drop_low_coverage
+            method: cnvkit_method
+            reference_cnn: cnvkit_reference_cnn
+            tumor_bam: index_bam/indexed_bam
```

Comment:
For Manta, I know we discussed running the tumor-only mode to use more sensitive parameters. This was required to identify a known SV. Do we have similar information to support running CNVkit in tumor-only mode? Is that why we are providing the single-sample input to the tumor_bam?

Reply:
My understanding of CNVkit is you're always comparing the tumor to a normal. There are 3 options for what you can use as the normal.
1) a paired normal bam
2) a generated reference.cnn (allowing pooling of normals)
3) "flat" coverage (don't need a normal)

Right now the CWLs do not fully reflect these requirements of running CNVkit. We will need to implement some exclusive and maybe dependent parameters to get the requirements of CNVkit in CWL. 
We are planning on using the reference.cnn which works in the current setup. And using a reference.cnn created by pooling normals together produces slightly better results than just running the matched tumor normal bams together.

Reply:
Sounds good. Let's be sure to review this in person at the Wednesday 10am.